fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -590,6 +596,9 @@ static Type fromObject(@Nullable final Object object) {
             }
             return new Type.Array(Type.fromListObject((List<?>)object));
         }
+        if (object instanceof DynamicMessage) {
+            return Record.fromDescriptor(((DynamicMessage) object).getDescriptorForType());
+        }
```

Comment:
This is change from Pranjal's PR: https://github.com/FoundationDB/fdb-record-layer/pull/2807/files#diff-0b60f460d35324d34656a59f5a7cc9b5660d4ee0f5a5c675946ba69f5ba07009

Reply:
Why is that in here?

Reply:
I'll remove it.